src/anomalib/metrics/per_image/_binclf_curve_numba.py

```diff
@@ -0,0 +1,117 @@
+"""Binary classification matrix curve (NUMBA implementation of low level functions).
+
+Details: `.binclf_curve`.
+
+author: jpcbertoldo
+"""
+
+# Original Code
+# https://github.com/jpcbertoldo/aupimo
+#
+# Modified
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import numba
```

Comment:
the pytorch-only was slower than the one with numpy (which is slower than numba)
i see this is becoming a long issue, i can just drop numba, but can you confirm this is the decision to make? (just to avoid going back later)

Reply:
@jpcbertoldo would it be possible to use the pytorch-only implementation by default, and use the Numba implementation if the user has Numba installed in their environment? This way we don't force casual users to install another dependency, and advanced users who care more about speed can install Numba to speed up the computation. We could log a warning to notify the users that the computation can be made faster by installing Numba.

Reply:
@djdameln this is already the case

this https://github.com/openvinotoolkit/anomalib/blob/440bf2f5f32c2a9230485a6df11ce3b026efa306/src/anomalib/metrics/per_image/binclf_curve_numpy.py#L23-L32 is checking if numba is installed and only importing the the numba-dependent version if it is the case

and this part https://github.com/openvinotoolkit/anomalib/blob/440bf2f5f32c2a9230485a6df11ce3b026efa306/src/anomalib/metrics/per_image/binclf_curve_numpy.py#L229-L239  is doing this version routing (with or without numba)

the installation of numba is already optional as well and it requires "opt-in" from the user: https://github.com/openvinotoolkit/anomalib/blob/440bf2f5f32c2a9230485a6df11ce3b026efa306/pyproject.toml#L87

@ashwinvaidya17 do you think that is ok like this?

Reply:
@samet-akcay can we give a final push to decide about keeping `numba` or not?
i think it is the only thing blocking the merge
the code doesn't depend on it, for the moment it is optional to install/use it, so the question is if we keep it or just get rid of it